tensorflow/core/util/gpu_kernel_helper.h

```diff
@@ -56,8 +56,14 @@ using gpuError_t = hipError_t;
 // macro wrapper to declare dynamic shared memory
 #if GOOGLE_CUDA
 
-#define GPU_DYNAMIC_SHARED_MEM_DECL(ALIGN, TYPE, NAME) \
-  extern __shared__ __align__(ALIGN) TYPE NAME[]
+#if defined(__APPLE__)
```

Comment:
Simpy ignoring alignment is not going to work. There must be a way to align. Maybe with __attribute__?

Reply:
I just tested that
```cpp
#define GPU_DYNAMIC_SHARED_MEM_DECL(ALIGN, TYPE, NAME) \
  extern __shared__  __attribute__ ((aligned(ALIGN))) TYPE NAME[]
```
is OK on macOS. I'll changed the code accordingly.

Reply:
Unfortunately, the above declaration with alignment is only semantically good. It will lead to https://github.com/tensorflow/tensorflow/issues/14174 and trigger multiple compilation errors such as
```
tensorflow/core/kernels/concat_lib_gpu_impl.cu.cc(73): error: specified alignment (4) is different from alignment (2) specified on a previous declaration
          detected during:
            instantiation of "void tensorflow::concat_variable_kernel<T,IntType,useSmem>(tensorflow::GpuDeviceArrayStruct<const T *, 8>, tensorflow::GpuDeviceArrayStruct<IntType, 8>, IntType, IntType, T *) [with T=float, IntType=tensorflow::int32, useSmem=true]" 
(161): here
            instantiation of "void tensorflow::ConcatGPUImpl(const Eigen::GpuDevice &, const tensorflow::GpuDeviceArrayStruct<const T *, 8> &, const tensorflow::GpuDeviceArrayStruct<IntType, 8> &, __nv_bool, int, tensorflow::TTypes<T, 2, Eigen::DenseIndex>::Matrix *) [with T=float, IntType=tensorflow::int32]" 
(224): here
```